src/migration/1717589112159-SourceStatsTrigger.ts

```diff
@@ -0,0 +1,128 @@
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER) , 0) + 1))
+          WHERE   id = NEW."sourceId";
+          RETURN NEW;
+        END;
+        $$
+      `);
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION update_squad_upvotes_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(
+                            flags,
+                            '{totalUpvotes}',
+                            to_jsonb(
+                              GREATEST(
+                                0,
+                                COALESCE(CAST(flags->>'totalUpvotes' AS INTEGER), 0) + (NEW.upvotes - OLD.upvotes)
+                              )
+                            )
+                          )
+          WHERE   id = NEW."sourceId";
+          RETURN NEW;
+        END;
+        $$
+      `);
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_views_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(
+                            flags,
+                            '{totalViews}',
+                            to_jsonb(COALESCE(CAST(flags->>'totalViews' AS INTEGER), 0) + (NEW.views - OLD.views))
+                          )
+          WHERE   id = NEW."sourceId";
+          RETURN NEW;
+        END;
+        $$
+      `);
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER) , 0) + 1))
+          WHERE   id = NEW."sourceId"
+          AND     type = 'squad';
+          RETURN NEW;
+        END;
+        $$
+      `);
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION update_squad_upvotes_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(
+                            flags,
+                            '{totalUpvotes}',
+                            to_jsonb(
+                              GREATEST(
+                                0,
+                                COALESCE(CAST(flags->>'totalUpvotes' AS INTEGER), 0) + (NEW.upvotes - OLD.upvotes)
+                              )
+                            )
+                          )
+          WHERE   id = NEW."sourceId"
+          AND     type = 'squad';
+          RETURN NEW;
+        END;
+        $$
+      `);
```

Comment:
For the rollback, we just need to copy the initial function:

https://github.com/dailydotdev/daily-api/blob/722a3016a868efeda40cd1b0629a825483ff3823/src/migration/1714662136871-SquadTotalUpvotesTrigger.ts#L9-L32